idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/editor/folding/FoldingSyntax.kt

```diff
@@ -1,48 +1,34 @@
 package arrow.meta.ide.dsl.editor.folding
 
 import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.phases.editor.extension.ExtensionProvider
 import arrow.meta.phases.ExtensionPhase
 import com.intellij.lang.ASTNode
 import com.intellij.lang.Language
 import com.intellij.lang.folding.FoldingBuilder
-import com.intellij.lang.folding.FoldingBuilderEx
 import com.intellij.lang.folding.FoldingDescriptor
-import com.intellij.lang.folding.LanguageFolding
 import com.intellij.openapi.editor.Document
-import com.intellij.psi.PsiElement
-import com.intellij.util.KeyedLazyInstance
 import org.jetbrains.kotlin.idea.KotlinLanguage
 
 interface FoldingSyntax {
-  fun IdeMetaPlugin.addFoldingBuilder(
+
+  fun IdeMetaPlugin.addFoldingBuilder(lang: Language = KotlinLanguage.INSTANCE, foldingBuilder: FoldingBuilder): ExtensionPhase =
+    ExtensionProvider.AddFoldingExtension(lang, foldingBuilder)
+
+  fun IdeMetaPlugin.createFoldingBuilder(
     placeHolderText: (node: ASTNode) -> String?,
-    foldRegions: (root: PsiElement, document: Document, quick: Boolean) -> List<FoldingDescriptor>,
+    buildFoldRegions: (node: ASTNode, document: Document) -> Array<FoldingDescriptor>,
```

Comment:
keep the names from before. 
```suggestion
    foldRegions: (node: ASTNode, document: Document) -> List<FoldingDescriptor>,
```